src/yb/yql/pgwrapper/pg_wrapper.cc

```diff
@@ -349,14 +357,37 @@ Status PgWrapper::Start() {
   }
 
   argv.push_back("-c");
-  // TODO: we should probably load the metrics library in a different way once we let
-  // users change the shared_preload_libraries conf parameter.
+
+  // Gather the default extensions:
+  vector<string> metricsLibs;
   if (FLAGS_pg_stat_statements_enabled) {
-    argv.push_back("shared_preload_libraries=pg_stat_statements,yb_pg_metrics,pgaudit,"
-      "pg_hint_plan");
+    metricsLibs.push_back("pg_stat_statements");
+    metricsLibs.push_back("yb_pg_metrics");
+    metricsLibs.push_back("pgaudit");
+    metricsLibs.push_back("pg_hint_plan");
   } else {
-    argv.push_back("shared_preload_libraries=yb_pg_metrics,pgaudit,pg_hint_plan");
+    metricsLibs.push_back("pg_stat_statements");
+    metricsLibs.push_back("pgaudit");
+    metricsLibs.push_back("pg_hint_plan");
   }
+
+  // Once again parse the pg conf:
+  vector<string> pgConfLines;
+  if (!FLAGS_ysql_pg_conf_csv.empty()) {
+    RETURN_NOT_OK(ReadCSVValues(FLAGS_ysql_pg_conf_csv, &pgConfLines));
+  } else if (!FLAGS_ysql_pg_conf.empty()) {
+    ReadCommaSeparatedValues(FLAGS_ysql_pg_conf, &pgConfLines);
+  }
+  
+  // If the user has given any shared_preload_libraries -> merge them in
+  for ( string &value : pgConfLines ) {
```

Comment:
nit: no space after (, no space before )

Reply:
Done in https://github.com/yugabyte/yugabyte-db/pull/9576/commits/dc53f3c3b497f605e6dd7a97a2a03b12ed581d37.